src/app/icd/client/DefaultICDClientStorage.cpp

```diff
@@ -462,10 +462,28 @@ CHIP_ERROR DefaultICDClientStorage::DeleteAllEntries(FabricIndex fabricIndex)
     return mpClientInfoStore->SyncDeleteKeyValue(DefaultStorageKeyAllocator::FabricICDClientInfoCounter(fabricIndex).KeyName());
 }
 
-CHIP_ERROR DefaultICDClientStorage::ProcessCheckInPayload(const ByteSpan & payload, ICDClientInfo & clientInfo)
+CHIP_ERROR DefaultICDClientStorage::ProcessCheckInPayload(const ByteSpan & payload, ICDClientInfo & clientInfo,
+                                                          CounterType & counter)
 {
-    // TODO: Need to implement default decription code using CheckinMessage::ParseCheckinMessagePayload
-    return CHIP_NO_ERROR;
+    /*appDataBuffer is the working buffer that will be used to retrieve data from the payload.
+     *counter to retrieve - 4 bytes
+     *appData to retrieve - 2 bytes(activeModeThreshold)
+     */
```

Comment:
Why are these comments here?  Shouldn't they be next to the definition of kAppDataLength?

Reply:
It is added next to the definition, as well. I added this to provide more clarity about the appDataBuffer.

Reply:
The sizing info does not belong here, because it has nothing to do with anything in this code.

Reply:
Okay. Will remove it from here.